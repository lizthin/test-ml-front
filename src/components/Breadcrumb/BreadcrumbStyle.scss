@import "../../assets/styles/colors.scss";
@import "../../assets/styles/mixins.scss";


.flex-categories {
    display: flex;
    float: left;
    padding-right: 5px;
    cursor: pointer;
    @include for-phone-only {
        display: none;
    }
}

.p-categories {
    color: $gray-dark;
    font-size: 14px;
    float: left;
}
