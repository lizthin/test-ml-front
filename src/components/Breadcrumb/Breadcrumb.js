import React from "react";
import PropTypes from "prop-types";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faChevronRight } from '@fortawesome/free-solid-svg-icons'
import "./BreadcrumbStyle.scss";

/**
 * Funtion component Breadcrumb print categories.
 * @param {object} properties props router, category list
 */
const Breadcrumb = ({
  properties: { category, lengthCatgories, index, history },
}) => {

/**
 * function handleRedirectUlr change history url when clicked breadcrumb category
 * @param {array} category response filter enabled
 * @example handleRedirectUrl("bolsos")
 */
  const handleRedirectUrl = (category) => {
    history.push(`/items?search=${category.trim()}`);
  };

  return (
    <div className="flex-categories" key={category}>
      <p className="p-categories"
        onClick={() => {
          handleRedirectUrl(category);
        }}
      >
        {category}
      </p>
      {(index + 1) < lengthCatgories ? <FontAwesomeIcon icon={faChevronRight} style={{
        fontSize: 19,
        color: '#999999',
        float: 'left',
        alignSelf: 'center',
        paddingLeft: 5,
        paddingRigth: 5,
      }} /> : <></>}
    </div>
  );
}

Breadcrumb.propTypes = {
  properties: PropTypes.object.isRequired,
};
export default Breadcrumb;