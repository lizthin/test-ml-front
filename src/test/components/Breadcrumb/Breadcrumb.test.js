import React from "react";
import Breadcrumb from "../../../components/Breadcrumb/Breadcrumb";
import { shallow } from "enzyme";

describe("should create a breadcrumb whit categories", () => {
  const mockProperties = {
    category:"Ropa y Calzado",
    handleRedirectUrl:jest.fn(),
    index:5,
    lengthCatgories:5,
  };

  test("should create component Breadcrumb", () => {
    const wrapper = shallow(<Breadcrumb properties={mockProperties} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('should call handleRedirectUrl', () => {
    const wrapper = shallow(<Breadcrumb properties={mockProperties} />);
    wrapper.find('.p-categories').at(0).simulate('click');
    const newp = wrapper.find('p').at(0).text();
    expect(newp).toBe('Ropa y Calzado')
    
  })
  
  test('should not create a breadcrumb', () => {
    const mockPropertiesCategory = {
      category:"Ropa y Calzado",
      handleRedirectUrl:jest.fn(),
      index:3,
      lengthCatgories:5,
    };
    const wrapper = shallow(<Breadcrumb properties={mockPropertiesCategory} />);
    expect(wrapper).toMatchSnapshot();

  })
  
});
