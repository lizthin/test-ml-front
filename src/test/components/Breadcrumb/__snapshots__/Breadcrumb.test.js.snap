// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should create a breadcrumb whit categories should create component Breadcrumb 1`] = `
<div
  className="flex-categories"
  key="Ropa y Calzado"
>
  <p
    className="p-categories"
    onClick={[Function]}
  >
    Ropa y Calzado
  </p>
</div>
`;

exports[`should create a breadcrumb whit categories should not create a breadcrumb 1`] = `
<div
  className="flex-categories"
  key="Ropa y Calzado"
>
  <p
    className="p-categories"
    onClick={[Function]}
  >
    Ropa y Calzado
  </p>
  <FontAwesomeIcon
    border={false}
    className=""
    fixedWidth={false}
    flip={null}
    icon={
      Object {
        "icon": Array [
          320,
          512,
          Array [],
          "f054",
          "M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z",
        ],
        "iconName": "chevron-right",
        "prefix": "fas",
      }
    }
    inverse={false}
    listItem={false}
    mask={null}
    pull={null}
    pulse={false}
    rotation={null}
    size={null}
    spin={false}
    style={
      Object {
        "alignSelf": "center",
        "color": "#999999",
        "float": "left",
        "fontSize": 19,
        "paddingLeft": 5,
        "paddingRigth": 5,
      }
    }
    swapOpacity={false}
    symbol={false}
    title=""
    transform={null}
  />
</div>
`;
